libsolidity/interface/CompilerStack.cpp

```diff
@@ -1251,12 +1264,45 @@ void CompilerStack::generateIR(ContractDefinition const& _contract)
 	tie(compiledContract.yulIR, compiledContract.yulIROptimized) = generator.run(_contract, otherYulSources);
 }
 
+void CompilerStack::generateEVMFromIR(ContractDefinition const& _contract)
+{
+	solAssert(m_stackState >= AnalysisPerformed, "");
+	if (m_hasError)
+		BOOST_THROW_EXCEPTION(CompilerError() << errinfo_comment("Called generateEVM with errors."));
+
+	if (!_contract.canBeDeployed())
+		return;
+
+	Contract& compiledContract = m_contracts.at(_contract.fullyQualifiedName());
+	solAssert(!compiledContract.yulIROptimized.empty(), "");
+	if (!compiledContract.object.bytecode.empty())
```

Comment:
When would this not be empty?

Reply:
If you call it on an already compiled contract. Copy from `generateEwasm`.

Reply:
When would that be the case though? Just wondering whether this should be an assertion instead

Reply:
`compileContract` for the current pipeline does the same (`if (_otherCompilers.count(&_contract)) return`). I think there is some case where through a dependency tree it would try to compile it multiple times, I recall something like that so this was added deliberately.

Reply:
Hm ok